mindsdb/integrations/handlers/langchain_handler/mindsdb_chat_model.py

```diff
@@ -0,0 +1,180 @@
+    CallbackManagerForLLMRun,
+)
+from langchain_core.language_models.chat_models import (
+    BaseChatModel,
+)
+from langchain_core.messages import (
+    AIMessage,
+    BaseMessage,
+    ChatMessage,
+    FunctionMessage,
+    HumanMessage,
+    SystemMessage,
+)
+from langchain_core.outputs import (
+    ChatGeneration,
+    ChatResult,
+)
+from langchain_core.pydantic_v1 import root_validator
+
+logger = logging.getLogger(__name__)
+
+
+def _convert_message_to_dict(message: BaseMessage) -> dict:
+    if isinstance(message, ChatMessage):
+        message_dict = {"role": message.role, "content": message.content}
+    elif isinstance(message, HumanMessage):
+        message_dict = {"role": "user", "content": message.content}
+    elif isinstance(message, AIMessage):
+        message_dict = {"role": "assistant", "content": message.content}
+        if "function_call" in message.additional_kwargs:
+            message_dict["function_call"] = message.additional_kwargs["function_call"]
+    elif isinstance(message, SystemMessage):
+        message_dict = {"role": "system", "content": message.content}
+    elif isinstance(message, FunctionMessage):
+        message_dict = {
+            "role": "function",
+            "content": message.content,
+            "name": message.name,
+        }
+    else:
+        raise ValueError(f"Got unknown type {message}")
+    if "name" in message.additional_kwargs:
+        message_dict["name"] = message.additional_kwargs["name"]
+    return message_dict
+
+
+class ChatMindsdb(BaseChatModel):
+    """A chat model that uses the Mindsdb"""
+
+    model_name: str
+    project_name: Optional[str] = 'mindsdb'
+    model_info: Optional[dict] = None
+    output_col: Optional[str] = None
+    project_datanode: Optional[Any] = None
+
+    @property
+    def _default_params(self) -> Dict[str, Any]:
+        return {}
+
+    def completion(
+        self, messages
+    ) -> Any:
+
+        model_info = self.model_info
+
+        # TODO create table for conversational model?
+        if len(messages) > 1:
+            content = '\n'.join([
+                f"{m['role']}: {m['content']}"
+                for m in messages
+            ])
+        else:
+            content = messages[0]['content']
+
+        record = {}
+        params = {}
+        if model_info.get('mode') == 'conversational':
+            user_column = model_info['user_column']
+            record[user_column] = content
+        else:
+            params['prompt_template'] = content
+
+        predictions = self.project_datanode.predict(
+            model_name=self.model_name,
+            data=[record],
+            params=params,
+        )
+
+        col = self.output_col
+        if col not in predictions.columns:
+            # get first column
+            col = predictions.columns[0]
+
+        # get first row
+        result = predictions[col][0]
+
+        # TODO token calculation
+        return {
+            'messages': [result]
```

Comment:
I kept it without usage calculation because it is topic for discuss. It could be done in ways:
- modify llm handlers and add usage statistics in model output 
- calculate it indirectly, for example used_tokens = ( len(input) + len(output) ) * K